pubsub/kafkapubsub/kafka_test.go

```diff
@@ -499,6 +499,11 @@ func TestOpenSubscriptionFromURL(t *testing.T) {
 	}{
 		// OK, but still error because broker doesn't exist.
 		{"kafka://mygroup?topic=mytopic", true},
+		// OK, specifying initial offset, but still error because broker doesn't exist.
```

Comment:
These tests are terrible (not your fault), since they all expect an error. What do you think about checking more explicitly for the "invalid query parameter" error so that we can distinguish the "good" URL parameters from the bad ones?

Reply:
Yeah I can do that. How about modifying the tests so you can specify the type of error you get back? Effectively ignoring the "has your client ran out of brokers to talk to?" one, since that will happen every time?

Reply:
Not sure exactly what you mean by "type of error", but that sounds right.

(I was thinking of changing `WantErr` to be a `string` instead of a `bool`, and the test would check that the returned `err.Error()` had that string in it. Checking error strings is generally a bad idea, but I think it would improve the coverage here.

Reply:
That's what I meant, using `type` was the wrong word since we have actual types. I'm just talking about string comparisons. I'll double check what sarama returns isn't actually a sentinel error but it might just come down to comparing strings.